plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7623,6 +7624,110 @@ public void testSystemTables()
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testShowCreateWithExtraProperties()
+    {
+        String tableName = format("%s.%s.show_create_table_with_extra_properties_%s", getSession().getCatalog().get(), getSession().getSchema().get(), randomNameSuffix());
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        // Assert that SHOW CREATE TABLE does not contain extra_properties
+        assertThat((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue())
+                .satisfies(new Condition<>(
+                        queryResult -> queryResult.contains("extra_properties"), "noExtraProperties"
+                ));
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testDuplicateExtraProperties()
+    {
+        assertQueryFails(
+                "CREATE TABLE create_table_with_duplicate_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false']))",
+                "Invalid value for catalog 'iceberg' table property 'extra_properties': Cannot convert.*");
+
+        assertQueryFails(
+                "CREATE TABLE create_table_as_with_duplicate_extra_properties WITH (extra_properties = MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false'])) AS SELECT 1 as c1",
+                "Invalid value for catalog 'iceberg' table property 'extra_properties': Cannot convert.*");
+    }
+
+    @Test
+    public void testNullExtraProperty()
+    {
+        assertQueryFails(
+                "CREATE TABLE create_table_with_duplicate_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['null.property'], ARRAY[null]))",
+                ".*Extra table property value cannot be null '\\{null.property=null}'.*");
+
+        assertQueryFails(
+                "CREATE TABLE create_table_with_as_duplicate_extra_properties WITH (extra_properties = MAP(ARRAY['null.property'], ARRAY[null])) AS SELECT 1 as c1",
+                ".*Extra table property value cannot be null '\\{null.property=null}'.*");
+    }
+
+    @Test
+    public void testAlterTableExtraProperty()
+    {
+        String tableName = "alter_table_with_extra_property_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer)".formatted(tableName));
+
+        assertUpdate("ALTER TABLE %s SET PROPERTIES extra_properties = MAP(ARRAY['extra.property.one'], ARRAY['one'])".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one')".formatted(tableName),
+                "VALUES 'one'");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
```

Comment:
let's not add yet this functionality. It has quite a bit of complexity - you can change /remove  relevant properties for the table and render it completely unusable when doing a seemingly simple property update.